docs/manual/src/internals/foreign_calls.md

```diff
@@ -0,0 +1,84 @@
+#[repr(C)]
+pub struct UniffiTodoListVTable {
+    // One function pointer field for each method
+    add: extern "C" fn(uniffi_handle: u64, title: RustBuffer, uniffi_out_return: &mut (), uniffi_call_status: &mut RustCallStatus),
+    get_items: extern "C" fn(uniffi_handle: u64, uniffi_out_return: &mut RustBuffer, uniffi_call_status: &mut RustCallStatus);
+    // A function pointer for freeing the callback interface
+    uniffi_free: extern "C" fn(uniffi_handle: u64);
+}
+
+/// A function to register the VTable
+#[no_mangle]
+pub extern "C" fn uniffi_init_todo_list_vtable(vtable: std::ptr::NonNull<UniffiTodoListVTable>) {...}
+```
+
+
+## VTable methods
+
+VTable methods work similar to a [Rust method call](./rust_calls.md)], with some exceptions.
+
+Instead of returning the value as normal, it's written to an out pointer.
+This is because some foreign languages, like Python, have issues returning C structs like `RustBuffer`.
+Methods that don't return anything still have an out pointer argument, but it's never written to.
+
+The first argument is a `u64` handle rather than a `void *`.
+In both cases however, it's a handle that represents a reference the underlying object.
+These are different for historical reasons only, we hope to eventually use the same type for both cases.
+
+Different bindings have different systems for callback interface handles, but they generally fall into 2 camps:
+* The handle is the key to a hash map that maps `u64` -> callback interface objects.
+* The is a raw pointer to a callback interface object casted to a `u64`.
+
+The method call is handled very similarly to a Rust method call:
+
+* If the function returns successfully then the lowered return value is written to `uniffi_out_return` and nothing is written to `uniffi_call_status`.
+* If the function throws the exception that corresponds to the `Result::Err` side of the Rust function, then:
+  * The foreign bindings set `uniffi_call_status.code` to `RustCallStatusCode::Error`
+  * The foreign bindings serialize the error into a `RustBuffer` and write it to `uniffi_call_status.error_buf`.
+  * Nothing is written to `uniffi_out_return`.
+* If the method throws some other exception, then:
+  * The foreign bindings set `uniffi_call_status.code` to `RustCallStatusCode::InternalError`
+  * The foreign bindings should try to to serialize the error message to a `RustBuffer` and write it to `error_buf`.
+    However, bindings may also not write to `error_buf` and the generated Rust code will handle this.
+  * Nothing is written to `uniffi_out_return`.
+  * If the callback method returns a `Result<T, E>` type and the bindings implement `From<uniffi::UnexpectedUniFFICallbackError> for E`,
+    then the generated Rust will convert the error to an `Err(E)` value and return it.
+  * If not, then the Rust bindings will panic.
+  * This is why it's recommended to always have callback methods return `Result` types and implement `From<uniffi::UnexpectedUniFFICallbackError>`.
```

Comment:
Maybe combine this sentence with the above (ie, it's part of the point above rather than a new point)